extensions/git/src/repository.ts

```diff
@@ -1649,48 +1649,48 @@ export class Repository implements Disposable {
 		}
 
 		await this.run(Operation.Sync, async () => {
-			await this.maybeAutoStash(async () => {
-				const config = workspace.getConfiguration('git', Uri.file(this.root));
-				const fetchOnPull = config.get<boolean>('fetchOnPull');
-				const tags = config.get<boolean>('pullTags');
-				const followTags = config.get<boolean>('followTagsWhenSync');
-				const supportCancellation = config.get<boolean>('supportCancellation');
-
-				const fn = async (cancellationToken?: CancellationToken) => {
-					// When fetchOnPull is enabled, fetch all branches when pulling
-					if (fetchOnPull) {
-						await this.fetchAll({}, cancellationToken);
-					}
+			const config = workspace.getConfiguration('git', Uri.file(this.root));
+			const fetchOnPull = config.get<boolean>('fetchOnPull');
+			const tags = config.get<boolean>('pullTags');
+			const followTags = config.get<boolean>('followTagsWhenSync');
+			const supportCancellation = config.get<boolean>('supportCancellation');
+			const autoStash = config.get<boolean>('autoStash');
 
-					if (await this.checkIfMaybeRebased(this.HEAD?.name)) {
-						await this._pullAndHandleTagConflict(rebase, remoteName, pullBranch, { tags, cancellationToken });
-					}
+			const fn = async (cancellationToken?: CancellationToken) => {
+				// When fetchOnPull is enabled, fetch all branches when pulling
+				if (fetchOnPull) {
+					await this.fetchAll({}, cancellationToken);
+				}
+
+				if (await this.checkIfMaybeRebased(this.HEAD?.name)) {
+					await this._pullAndHandleTagConflict(rebase, remoteName, pullBranch, { tags, cancellationToken, autoStash, });
```

Comment:
btw I also see these options duplicated a few times why not encapsulate resolving these options into inner function?